src/js/containers/account/visualizations/AccountTimeVisualizationContainer.jsx

```diff
@@ -153,8 +223,93 @@ export class AccountTimeVisualizationSectionContainer extends React.Component {
         const xSeries = [];
         const ySeries = [];
         const allY = [];
-        if (this.state.hasFilteredObligated) {
-            // In progress
+        const hasFilteredObligated = (this.props.reduxFilters.objectClass.count() > 0)
+            || (this.props.reduxFilters.programActivity.count() > 0);
+
+        if (hasFilteredObligated) {
+            if (this.state.visualizationPeriod === 'quarter') {
```

Comment:
Simplified `fetchData()` and `parseData()` logic by checking for `visualizationPeriod` and `hasFilteredObligated` only where necessary and eliminating repetitive if-else blocks.